File: backend/app/schemas/instructor_assignment.py
from datetime import datetime
from typing import Optional

from pydantic import BaseModel


class AssignmentCreate(BaseModel):
    topic: str
    topic_type: str  # letter, motion_sign, word_sign


class AssignmentCompleteRequest(BaseModel):
    completed: bool = True


class AssignmentResponse(BaseModel):
    id: str
    learner_id: str
    instructor_id: str
    instructor_name: Optional[str] = None
    topic: str
    topic_type: str
    notes: Optional[str] = None
    due_date: Optional[str] = None  # "YYYY-MM-DD"
    completed: bool = False
    completed_at: Optional[datetime] = None
    # Populated only when the instructor attached a reference photo/video.
    # reference_media_url is a ready-to-use /media/... path (already
    # combined with the stored relative path) so the frontend never has
    # to know the storage layout — see media_upload_service.
    reference_media_url: Optional[str] = None
    reference_media_type: Optional[str] = None  # image, video
    created_at: Optional[datetime] = None

    class Config:
        from_attributes = True


class AssignmentListResponse(BaseModel):
    assignments: list[AssignmentResponse]

File: frontend/src/types/instructorAssignment.ts
// Mirrors backend/app/schemas/instructor_assignment.py

export type AssignmentTopicType = 'letter' | 'motion_sign' | 'word_sign';

export type AssignmentMediaType = 'image' | 'video';

export interface Assignment {
  id: string;
  learner_id: string;
  instructor_id: string;
  instructor_name: string | null;
  topic: string;
  topic_type: AssignmentTopicType;
  notes: string | null;
  due_date: string | null; // "YYYY-MM-DD"
  completed: boolean;
  completed_at: string | null;
  // Set only when the instructor attached a reference photo/video —
  // reference_media_url is a ready-to-fetch /media/... path.
  reference_media_url: string | null;
  reference_media_type: AssignmentMediaType | null;
  created_at: string | null;
}

export interface AssignmentListResponse {
  assignments: Assignment[];
}
